hybrid/decomposers.py

```diff
@@ -44,6 +45,91 @@ class IdentityDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
     def next(self, state, **runopts):
         return state.updated(subproblem=state.problem)
 
+class ComponentDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
+    """Selects a subproblem of variables that make up a connected component.
+
+    Args:
+        rolling (bool, optional, default=True):
+            If True, successive calls for the same problem (with possibly
+            different samples) produce subproblems on different components,
+            selected by rolling down the list of all components sorted by
+            the `key` function.
+
+        key (function, optional, default=len):
+            Function used to extract a comparison key from each component of 
+            the problem. The comparison keys determine which component will be 
+            used to build the subproblem. By default, the components are sorted 
+            by size in decreasing order, so the largest component will be used first.
+        
+        reverse (bool, optional, default=True):
+            If True, the components' comparison keys, found using the `key` function, 
+            will be sorted in decreasing order. If False, they will be sorted
+            in increasing order. 
+
+        silent_rewind (bool, optional, default=True):
+            If False, raises :exc:`EndOfStream` when resetting/rewinding the
+            subproblem generator once all components have been used. 
+    
+    See :ref:`decomposers-examples`.
+
+    """
+
+    def __init__(self, rolling=True, silent_rewind=True, key=len, reverse=True, **runopts):
```

Comment:
Common python gotcha is [using mutable objects for default values](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) of keyword arguments (`key=len`).